ui/app/src/modules/Preview/PreviewConcierge.tsx

```diff
@@ -175,15 +184,19 @@ export function PreviewConcierge(props: any) {
       }
     });
 
+    // Retrieve all content types in the system
+    (!contentTypes && site) && dispatch(fetchContentTypes());
+    contentTypes && contentTypes$.next(contentTypes);
+
     let fetchSubscription;
     switch (selectedTool) {
       case 'craftercms.ice.assets':
-        // TODO: aaron to fetch assets here...
+        (assets.isFetching === null && site) && dispatch(fetchPanelAssetsItems(initialSearchParameters));
```

Comment:
When switching sites, are you resetting the state so the isFetching is null? Test site switching works as expected. The preview toolbar switcher is great for testing this.